spec/client_spec.js

```diff
@@ -419,15 +419,15 @@ describe('Client', function() {
         it('resolves when the request succeeds', function(done) {
           triggerEvent(subject, 'request:' + requestsCount + '.done', response);
           promise.then(function() {
-            expect(doneHandler).to.have.been.calledWith(response.responseArgs[0]);
```

Comment:
@danielbreves That's a good point. I had a look again at the different of arguments recieved by a v1 and v2 app and saw a clear difference:

V1:
<img width="651" alt="screen shot 2017-05-31 at 17 10 56" src="https://cloud.githubusercontent.com/assets/5171412/26619972/28702bc0-4624-11e7-924f-0ccf4c0bafea.png">

V2:
<img width="853" alt="screen shot 2017-05-31 at 17 13 09" src="https://cloud.githubusercontent.com/assets/5171412/26620033/76f587e0-4624-11e7-89fe-55992ce97400.png">

So what happens is that obviously `ajax.done` is returning the same arguments (3). But it's then send through `postMessage` and it strips that first argument away, cause that's a `#document` node. The second, a string, passes along just fine. The third, a object with some string and a bunch of functions, gets stripped down to only the strings and a number.

This also completely makes sense why it does work when using `data: text` because the first argument stays a string and doesn't get processed.